Index: mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/gemm/bli_gemm_ker_var2.c
===================================================================
--- mdk_release_17.04.5_general_purpose.orig/mdk/common/components/BLIS/leon/frame/3/gemm/bli_gemm_ker_var2.c
+++ mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/gemm/bli_gemm_ker_var2.c
@@ -257,10 +257,6 @@ void bli_sgemm_ker_var2(
     for (int s = 0; s < BLIS_SHAVES; s++)
         shave_wait(blis_shaves[s]);
 
-#ifdef MYRIAD2
-    DrvShaveL2CachePartitionFlushAndInvalidate(0);
-#endif
-
 #ifdef SHAVE_PERFORMANCE_ONLY
     time_shave += (bli_clock() - countCycles);
 #endif
Index: mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_ll_ker_var2.c
===================================================================
--- mdk_release_17.04.5_general_purpose.orig/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_ll_ker_var2.c
+++ mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_ll_ker_var2.c
@@ -260,10 +260,6 @@ void bli_strsm_ll_ker_var2(
     for (int s = 0; s < BLIS_SHAVES; s++)
         shave_wait(blis_shaves[s]);
 
-#ifdef MYRIAD2
-    DrvShaveL2CachePartitionFlushAndInvalidate(0);
-#endif
-
 #ifdef SHAVE_PERFORMANCE_ONLY
     time_shave += (bli_clock() - countCycles);
 #endif
Index: mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_lu_ker_var2.c
===================================================================
--- mdk_release_17.04.5_general_purpose.orig/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_lu_ker_var2.c
+++ mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_lu_ker_var2.c
@@ -270,10 +270,6 @@ void bli_strsm_lu_ker_var2(
     for (int s = 0; s < BLIS_SHAVES; s++)
         shave_wait(blis_shaves[s]);
 
-#ifdef MYRIAD2
-    DrvShaveL2CachePartitionFlushAndInvalidate(0);
-#endif
-
 #ifdef SHAVE_PERFORMANCE_ONLY
     time_shave += (bli_clock() - countCycles);
 #endif
Index: mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_rl_ker_var2.c
===================================================================
--- mdk_release_17.04.5_general_purpose.orig/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_rl_ker_var2.c
+++ mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_rl_ker_var2.c
@@ -275,10 +275,6 @@ void bli_strsm_rl_ker_var2(
     for (int s = 0; s < BLIS_SHAVES; s++)
         shave_wait(blis_shaves[s]);
 
-#ifdef MYRIAD2
-    DrvShaveL2CachePartitionFlushAndInvalidate(0);
-#endif
-
 #ifdef SHAVE_PERFORMANCE_ONLY
     time_shave += (bli_clock() - countCycles);
 #endif
Index: mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_ru_ker_var2.c
===================================================================
--- mdk_release_17.04.5_general_purpose.orig/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_ru_ker_var2.c
+++ mdk_release_17.04.5_general_purpose/mdk/common/components/BLIS/leon/frame/3/trsm/bli_trsm_ru_ker_var2.c
@@ -279,10 +279,6 @@ void bli_strsm_ru_ker_var2(
     for (int s = 0; s < BLIS_SHAVES; s++)
         shave_wait(blis_shaves[s]);
 
-#ifdef MYRIAD2
-    DrvShaveL2CachePartitionFlushAndInvalidate(0);
-#endif
-
 #ifdef SHAVE_PERFORMANCE_ONLY
     time_shave += (bli_clock() - countCycles);
 #endif
